.github/scripts/build.functions_tests.sh

```diff
@@ -4,7 +4,9 @@ set -eu
 
 SCRIPT_DIR="$(dirname "$(readlink -f "$0")")"
 
-. "$SCRIPT_DIR"/assert.sh/assert.sh
+assert_script_content=$(curl --silent https://raw.githubusercontent.com/hazelcast/assert.sh/main/assert.sh)
+# shellcheck source=/dev/null
+. <(echo "${assert_script_content}")
```

Comment:
Can you explain what these two lines do?
```
# shellcheck source=/dev/null
. <(echo "${assert_script_content}")
```
Also, what happens if `curl` fails?

Reply:
Currently we source (`.`) the content of another script on the local filesystem into the environment.

This change:
- downloads the latest copy of the script from our fork, saving the content to a variable
- feeds that into the source mechanism, as if the variable content was actually a file ([more info](https://superuser.com/a/1060002/1220887))

Reply:
> Also, what happens if curl fails?

Sorry, missed that. 

_Should_ fail, because the script has `set -e`.
That's the reason for splitting out download from source mechanism, to ensure non-zero exit codes are captured.

I've just tested it with a wrong URL, it does give a non-zero error code from the `test_scripts.sh` script - which means the GitHub actions using it will fail.
No error message, though.

Reply:
> Interesting but little confusing must say. Can we add a comment OR simply create file in the same location as before (pass location to curl)?

I'm not saying it's super readable as it is now (mostly because `bash`), but I'm not sure what comment would help comprehension. Did you have any ideas?

Reply:
May be point to https://github.com/hazelcast/hazelcast-docker/pull/779#discussion_r1668846294?
Otherwise its ok as I understand it now!

Reply:
43f6296c68e0f911d84a11854e7c823b2660e708